src/traits.rs

```diff
@@ -61,3 +66,80 @@ pub trait Union<RHS = Self> {
     /// Build the union shape of self and the given shape.
     fn union(&self, &RHS) -> Self::Output;
 }
+
+/// Primitive with axis aligned bounding box
+pub trait HasAabb {
+    /// Bounding box type
+    type Aabb: Aabb + Clone + Union<Self::Aabb, Output = Self::Aabb> + Debug;
+
+    /// Get the bounding box of the primitive in local space coordinates.
+    fn get_bound(&self) -> Self::Aabb;
+}
+
+/// Minkowski support function for primitive
+pub trait SupportFunction {
+    /// Point type
+    type Point: EuclideanSpace;
+
+    /// Get the support point on the shape in a given direction.
+    ///
+    /// ## Parameters
+    ///
+    /// - `direction`: The search direction in world space.
+    /// - `transform`: The current local to world transform for this primitive.
+    ///
+    /// ## Returns
+    ///
+    /// Return the point that is furthest away from the origin, in the given search direction.
+    /// For discrete shapes, the furthest vertex is enough, there is no need to do exact
+    /// intersection point computation.
+    ///
+    /// ## Type parameters
+    ///
+    /// - `P`: Transform type
+    fn support_point<T>(
+        &self,
+        direction: &<Self::Point as EuclideanSpace>::Diff,
+        transform: &T,
+    ) -> Self::Point
+    where
+        T: Transform<Self::Point>;
+}
+
+/// Discrete intersection test on transformed primitive
+pub trait DiscreteTransformed<RHS> {
+    /// Point type for transformation of self
+    type Point: EuclideanSpace;
+
+    /// Intersection test for transformed self
+    fn intersects_transformed<T>(&self, _: &RHS, _: &T) -> bool
```

Comment:
no need to specify `_` in the argument names of trait methods, just omit them entirely, e.g. `(&self, &RHS, &T) -> bool`